catalog/CatalogRelation.hpp

```diff
@@ -397,6 +414,8 @@ class CatalogRelation : public CatalogRelationSchema {
   // Mutex for locking the index scheme.
   alignas(kCacheLineBytes) mutable SpinSharedMutex<false> index_scheme_mutex_;
 
+  mutable std::size_t num_tuples_;
```

Comment:
Remove `mutable` here.
The combination of a `mutable` field and its related `const` method are for some special situations, e.g. when the `mutable` field is an auxiliary state (mutex, value cache) and the related method _logically_ does not change the state of the object.

This StackOverflow thread gives more discussion: http://stackoverflow.com/questions/105014/does-the-mutable-keyword-have-any-purpose-other-than-allowing-the-variable-to